### ISSUE TITLE
Live: ABR fails with subdur

### ISSUE THREAD

**Issue by DanielMartine** [gpac/gpac#1032]
On Ubuntu 16.04
Displayed on video.js contrib dash
Browser Safari/Chrome

Hi there, I need a hand to identify how to fix this bug, i try to archive a live streaming with the command above:

> MP4Box -dash-live 2000 -subdur 4000 -profile dashavc264:live -mpd-refresh 4 -time-shift 16 -min-buffer 2000 -url-template -out manifest.mpd 1080set.mp4#video 720set.mp4#video 360set.mp4#video 1080set.mp4#audio 720set.mp4#audio 360set.mp4#audio

Now it's running on www.allibrante.com

-I use -dash-live of 2000 ms couse the GOP of encoding is every 2 seconds.
-I use min-buffer of 2000 ms as the -bufsize of the encoding is 2 second (4800 kbps * 2)
-I use -mpd-refresh to 4 second as the subdur (4000ms)
-I read that url-template for live is needed

The adaption sets are encoded with the following ffmpeg command line:

> ffmpeg -f concat -safe 0 -y -i list.txt -i ../logo/logo.png -c:a aac -b:a 384k -ar 48000 -ac 2 -c:v libx264 -x264opts keyint=50:min-keyint=50:no-scenecut -r 25 -b:v 4800k -maxrate 4800k -bufsize 9600k -filter_complex "[0:v][1:v]overlay=main_w-overlay_w-10:10,scale=1920:1080,setsar=1" -t $TOTDUR ../output/1080set.mp4

> ffmpeg -f concat -safe 0 -y -i list.txt -i ../logo/logo.png -c:a aac -b:a 256k -ar 48000 -ac 2 -c:v libx264 -x264opts keyint=50:min-keyint=50:no-scenecut -r 25 -b:v 2400k -maxrate 2400k -bufsize 4800k -filter_complex "[0:v][1:v]overlay=main_w-overlay_w-10:10,scale=1280x720,setsar=1" -t $TOTDUR ../output/720set.mp4

> ffmpeg -f concat -safe 0 -y -i list.txt -i ../logo/logo.png -c:a aac -b:a 128k -ar 48000 -ac 2 -c:v libx264 -x264opts keyint=50:min-keyint=50:no-scenecut -r 25 -b:v 800k -maxrate 800k -bufsize 1600k -filter_complex "[0:v][1:v]overlay=main_w-overlay_w-10:10,scale=640x360,setsar=1" -t $TOTDUR ../output/360set.mp4

I verified that the gop was aligned with ffmpeg and mp4box, it' correct.
If i try without -subdur flag, the ABR work properly: trying with chrome/performance to reduce the bandwith it is capable to switch when needed.
I use subdur to avoid using too much memory: I dont need to make all the segments available at same time.
Actually the streaming try to load the most expensive rapresentation over and over again.
May i ask a help to understand what is going on here?
Thanks a lot!
Dan

**Comment by DanielMartine** [comment]
This is the response from dash conformance tool:

```
### error: ftyp-1 
### The brand 'isom' can only be a compatible, not major, brand
'tkhd' alternateGroup must be 0 not 1
Validate_ES_Descriptor: ES_ID should be 0 not 2 in media tracks

WARNING: unknown sample table atom 'sgpd'
Brand 'lmsg' not found as a compatible brand for the last segment (number 3); violates Section 3.2.3. of Interoperability Point DASH264: If the MPD@type is equal to "dynamic" or if it includes MPD@profile attribute in-cludes "urn:mpeg:dash:profile:isoff-live:2011", then: if the Media Segment is the last Media Segment in the Representation, this Me-dia Segment shall carry the 'lmsg' compatibility brand
tfdt base media decode time 30523.984000 not equal to accumulated decode time 0.000000 for track 2 for the first fragment of the movie. This software does not handle incomplete presentations. Applying correction.
### error: 
### grouping_type roll in sbgp is not found for any sgpd in moof number 1
### error: 
### grouping_type roll in sbgp is not found for any sgpd in moof number 1
### error: 
### grouping_type roll in sbgp is not found for any sgpd in moof number 1
### error: 
### grouping_type roll in sbgp is not found for any sgpd in moof number 1
### error: 
### grouping_type roll in sbgp is not found for any sgpd in moof number 1
### error: 
### grouping_type roll in sbgp is not found for any sgpd in moof number 1
### error: 
### grouping_type roll in sbgp is not found for any sgpd in moof number 1
### error: 
[...]

[...]
### grouping_type roll in sbgp is not found for any sgpd in moof number 2
### error: 
### grouping_type roll in sbgp is not found for any sgpd in moof number 2
### error: 
### grouping_type roll in sbgp is not found for any sgpd in moof number 2
### error: 
### Buffer underrun conformance error: first (and only one reported here) for sample 1 of run 1 of track fragment 1 of fragment 1 of track id 2 (sample absolute file offset 1307, fragment absolute file offset 811, bandwidth: 245)
```

**Comment by jeanlf** [comment]
these should have been fixed in 0.9.0 DEV, please retry and reopen if needed
